go/vt/vtgate/executor.go

```diff
@@ -1128,7 +1128,7 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, comments sqlparser.
 	statement := stmt
 	bindVarNeeds := sqlparser.BindVarNeeds{}
 	if !sqlparser.IgnoreMaxPayloadSizeDirective(statement) && !isValidPayloadSize(query) {
-		return nil, vterrors.New(vtrpcpb.Code_RESOURCE_EXHAUSTED, "query payload size above threshold")
+		return nil, vterrors.New(vtrpcpb.Code_RESOURCE_EXHAUSTED, vterrors.ResourceExhaustedQueryPayloadThresholdErrMsg)
```

Comment:
Instead of this you can create sqlError here and then no changed would be needed in sql_error.go. It will return the sqlError as-is.

```
Example:
mysql.NewSQLError(mysql.ERBadDb, "42000", "Unknown database '%s'", keyspace)
```

Reply:
Ah, nice -- it wasn't clear that we wanted to change this for all callers or how the executor would deal with getting a sql error back.

Reply:
✅